SConstruct

```diff
@@ -548,6 +548,8 @@ build_dir = 'build/%s/%s' % (
 
 env['ROC_BINDIR'] = '#bin/%s' % host
 env['ROC_VERSION'] = open(env.File('#.version').path).read().strip()
+env['ROC_SHA'] = os.popen("cat .git/$(cat .git/HEAD | awk '{print $2}') | cut -b1-10").read().strip()
```

Comment:
Thanks for the PR! Yes, this is what I was thinking about.

I'd change the cat/awk/cut pipeline into a pure-Python code to avoid build-time dependency on these tools and to make it portable across non-unix platforms. We will support Windows some day and I usually try to avoid unix-specific code in our build scripts, to make this future porting easier. And also, IIRC, awk is not installed by default on some minimalistic Linux distros like Alpine.

(And if this can't be done in a one-liner, I'd extract it into a function in [parsers.py](https://github.com/roc-project/roc/blob/master/site_scons/site_tools/roc/parsers.py#L119), something like env.ParseGitHead).

Reply:
Good thinking.. updated things to be straight Python with file opens, and extracted to a function in parsers.py. 

I'm not sure about the Travis failures..
On my machine, head_ref is equal to "refs/heads/feature/malcops/git-sha-to-version-info" and then opening ".git/refs/heads/feature/malcops/git-sha-to-version-info" provides the SHA.

Reply:
That's because the HEAD is detached on Travis.

You can reproduce it like this:

```
$ git checkout HEAD^
...
$ git status
HEAD detached at 020f7b9
nothing to commit, working tree clean
$ cat .git/refs/heads/master
a16b575580bbe70f06cc998541a46770cabbd885
```

I think we should correctly handle both situations.

Reply:
Thanks for the hint!